packages/backend/src/registries/ConfigurationRegistry.ts

```diff
@@ -51,9 +52,32 @@ export class ConfigurationRegistry extends Publisher<ExtensionConfiguration> imp
       apiPort: this.#configuration.get<number>('apiPort') ?? API_PORT_DEFAULT,
       experimentalTuning: this.#configuration.get<boolean>('experimentalTuning') ?? false,
       modelUploadDisabled: this.#configuration.get<boolean>('modelUploadDisabled') ?? false,
+      showGPUPromotion: this.#configuration.get<boolean>('showGPUPromotion') ?? true,
     };
   }
 
+  async updateExtensionConfiguration(update: Partial<ExtensionConfiguration>): Promise<void> {
+    if (update.modelsPath !== undefined) {
+      await this.#configuration.update('models.path', update.modelsPath);
+    }
+    if (update.experimentalGPU !== undefined) {
+      await this.#configuration.update('experimentalGPU', update.experimentalGPU);
+    }
+    if (update.apiPort !== undefined) {
+      await this.#configuration.update('apiPort', update.apiPort);
+    }
+    if (update.experimentalTuning !== undefined) {
+      await this.#configuration.update('experimentalTuning', update.experimentalTuning);
+    }
+    if (update.modelUploadDisabled !== undefined) {
+      await this.#configuration.update('modelUploadDisabled', update.modelUploadDisabled);
+    }
+    if (update.showGPUPromotion !== undefined) {
+      await this.#configuration.update('showGPUPromotion', update.showGPUPromotion);
+    }
```

Comment:
I'm wondering it it could not be generic

like, for each field send to the updated configuration, you send the instruction as it looks like it's the same configuration property name except for modelsPath where it's models.path (but probably you can make it different